### PR TITLE
Added tests for buildapi.py

### PR DESCRIPTION
**Pull request by adusca**
This patch only adds tests for query_jobs_schedule, but I'm hoping that having the structure will help to add tests for new functions.

### CODE REVIEW

**Comment by armenzg** [comment]
Could you have a look at the Travis output?

----------------------------- Captured stdout call -----------------------------
Please enter your full LDAP email address:

**Comment by adusca** [comment]
Sorry, I didn't realize I would have to mock credentials for Travis. It's passing now.

File: test/test_buildapi.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    def test_call_second_time(self, get_credentials, valid_revision, get):
+        """Calling the function again should return us the results directly from cache."""
+        self.assertEquals(
+            buildapi.query_jobs_schedule("try", "146071751b1e"),
+            json.loads(GET_CONTENT))
+        # query_jobs_schedule should return its value directly from
+        # cache without calling get
+        assert get.call_count == 0
+
+    @patch('mozci.sources.buildapi.valid_revision', return_value=True)
```
Comment (armenzg) [inline test/test_buildapi.py:91]: Any reason this patching happens in a different order than the previous two tests?

**Comment by armenzg** [comment]
Feel free to merge when you're ready.

Thanks for providing this test coverage increases!

**Pull request merged**